### DISCUSSION TITLE
Mix and match a subset of tar_map() values

### DISCUSSION THREAD

**Discussion by liutiming** [ropensci/targets#409] [Help] [answered]
Hi @wlandau , would love to hear your thoughts on this

Existing tar_map:
```r
    tar_map(
      values = tibble::tibble(
          id_vec = rlang::syms(c("A", "B", "C))
      )
    , tar_target(target_1, function(id_vec ))
```

what is the best way to process target_1_B and target_1_C with target_1_A? e.g. join the B and C with A. 

One can start another tar_map with only target_1_B and target_1_C, but this may not be the most elegant solution given longer names and repeated typing?

**Comment by wlandau** [comment, marked as answer]
Sounds like a job for `tar_combine()`.

``` r
library(targets)
tar_script({
  library(tarchetypes)
  targets <- tar_map(
    values = list(id = rlang::syms(c("A", "B", "C"))),
    tar_target(x, fun(id))
  )
  list(
    targets,
    tar_combine(y, targets)
  )
})

tar_manifest(fields = command)
#> # A tibble: 4 x 2
#>   name  command                                      
#>   <chr> <chr>                                        
#> 1 x_B   fun(B)                                       
#> 2 x_C   fun(C)                                       
#> 3 x_A   fun(A)                                       
#> 4 y     vctrs::vec_c(x_A = x_A, x_B = x_B, x_C = x_C)

tar_script({
  library(tarchetypes)
  targets <- tar_map(
    values = list(id = rlang::syms(c("A", "B", "C"))),
    tar_target(x, fun(id))
  )
  list(
    targets,
    tar_combine(y, targets, command = bind_rows(!!!.x, .id = "name"))
  )
})

tar_manifest(fields = command)
#> # A tibble: 4 x 2
#>   name  command                                                     
#>   <chr> <chr>                                                       
#> 1 x_B   "fun(B)"                                                    
#> 2 x_C   "fun(C)"                                                    
#> 3 x_A   "fun(A)"                                                    
#> 4 y     "bind_rows(x_A = x_A, x_B = x_B, x_C = x_C, .id = \"name\")"
```

<sup>Created on 2021-04-13 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

**Reply by liutiming** [reply]
Thanks. I probably could have explained better. What would you recommend if I want to have a table of AB and AC (instead of ABC that will be produced by tar_combine)?

**Reply by wlandau** [reply]
In that case, maybe either a second `tar_map()` or 2 `tar_combine()`s: `tar_combine(y, targets[c(1, 2)], command = bind_rows(!!!.x, .id = "name"))`, `tar_combine(y, targets[c(1, 3)], command = bind_rows(!!!.x, .id = "name"))`.

**Reply by liutiming** [reply]
yup thanks! i was thinking of something along the line of changing/filtering value in the middle of a tar_map but i imagine it may be quite challenging to implement programmatically.
